waspls/src/Wasp/LSP/Handlers.hs

```diff
@@ -0,0 +1,123 @@
+module Wasp.LSP.Handlers
+  ( initializedHandler,
+    didOpenHandler,
+    didChangeHandler,
+    didSaveHandler,
+  )
+where
+
+import Control.Lens ((+~), (^.))
+import Control.Monad.Trans (lift)
+import Control.Monad.Trans.Except (throwE)
+import Data.Function ((&))
+import Data.Text (Text)
+import qualified Data.Text as T
+import Language.LSP.Server (Handlers, LspT)
+import qualified Language.LSP.Server as LSP
+import Language.LSP.Types
+import Language.LSP.Types.Lens
+import Language.LSP.VFS (virtualFileText)
+import qualified Wasp.Analyzer
+import qualified Wasp.Analyzer.AnalyzeError as WE
+import Wasp.Analyzer.Parser (Ctx (Ctx))
+import Wasp.Analyzer.Parser.SourceRegion (getRgnEnd, getRgnStart)
+import Wasp.LSP.State (HandlerM, ServerConfig, Severity (..))
+
+-- LSP notification and request handlers
+
+-- | "Initialized" notification is sent when the client is started. We don't
+-- have anything we need to do at initialization, but this is required to be
+-- implemented.
+initializedHandler :: Handlers HandlerM
+initializedHandler =
+  LSP.notificationHandler SInitialized $ \_ -> return ()
+
+-- | "TextDocumentDidOpen" is sent by the client when a new document is opened.
+-- This handler then runs the `diagnosticsHandler`.
```

Comment:
This second line of comment is redundant since it can easily be seen from the implementation. But maybe you can quickly explain what this means. So we basically go to analyze this new document?